From b24fc72969cea7ef55eeaf6d258c42b9759e1eb8 Mon Sep 17 00:00:00 2001
From: Bhushan Shah <bhush94@gmail.com>
Date: Mon, 22 Feb 2021 21:13:41 +0530
Subject: [PATCH] [platforms/drm] only enable output on key press not release

Otherwise if powerdevil for example is configured to turn screen off on
the power key press, we will turn it right back on when we get key
release event.

Suggested-by: David Edmundson <kde@davidedmundson.co.uk>
---
plugins/platforms/drm/drm_inputeventfilter.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/plugins/platforms/drm/drm_inputeventfilter.cpp b/plugins/platforms/drm/drm_inputeventfilter.cpp
index 933fe1a87..9744fb348 100644
--- a/plugins/platforms/drm/drm_inputeventfilter.cpp
+++ b/plugins/platforms/drm/drm_inputeventfilter.cpp
@@ -42,9 +42,11 @@ bool DpmsInputEventFilter::wheelEvent(QWheelEvent *event)
 
 bool DpmsInputEventFilter::keyEvent(QKeyEvent *event)
 {
-    Q_UNUSED(event)
-    notify();
-    return true;
+    if (event->type() == QKeyEvent::KeyPress) {
+        notify();
+        return true;
+    }
+    return false;
 }
 
 bool DpmsInputEventFilter::touchDown(qint32 id, const QPointF &pos, quint32 time)
-- 
GitLab

From 58bee8e38ac9ec6e222566c14bbff475313c49c1 Mon Sep 17 00:00:00 2001
From: Vlad Zahorodnii <vlad.zahorodnii@kde.org>
Date: Mon, 22 Feb 2021 18:11:06 +0200
Subject: [PATCH] Fix build

---
plugins/platforms/drm/drm_inputeventfilter.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/plugins/platforms/drm/drm_inputeventfilter.cpp b/plugins/platforms/drm/drm_inputeventfilter.cpp
index 9744fb348..63b11f1ac 100644
--- a/plugins/platforms/drm/drm_inputeventfilter.cpp
+++ b/plugins/platforms/drm/drm_inputeventfilter.cpp
@@ -11,6 +11,7 @@
 #include "wayland_server.h"
 
 #include <QApplication>
+#include <QKeyEvent>
 
 #include <KWaylandServer/seat_interface.h>
 
-- 
GitLab

From ae8e69b4bb70b8915064bc5ae404d039db1ca587 Mon Sep 17 00:00:00 2001
From: Bhushan Shah <bhush94@gmail.com>
Date: Mon, 22 Feb 2021 21:49:39 +0530
Subject: [PATCH] [platforms/drm] accept both keypress and keyrelease event in
 filter

We don't want to pass out either of key press or release event to other
filters.
---
 src/plugins/platforms/drm/drm_inputeventfilter.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/plugins/platforms/drm/drm_inputeventfilter.cpp b/plugins/platforms/drm/drm_inputeventfilter.cpp
index 63b11f1ac..9b9d803f0 100644
--- a/plugins/platforms/drm/drm_inputeventfilter.cpp
+++ b/plugins/platforms/drm/drm_inputeventfilter.cpp
@@ -45,9 +45,8 @@ bool DpmsInputEventFilter::keyEvent(QKeyEvent *event)
 {
     if (event->type() == QKeyEvent::KeyPress) {
         notify();
-        return true;
     }
-    return false;
+    return true;
 }
 
 bool DpmsInputEventFilter::touchDown(qint32 id, const QPointF &pos, quint32 time)
-- 
GitLab
